common/src/main/java/org/conscrypt/ConscryptEngine.java

```diff
@@ -442,7 +442,9 @@ public void closeOutbound() {
                 return;
             }
             if (isHandshakeStarted()) {
+                state = STATE_CLOSED;
```

Comment:
I think you are correct, although I don't pretend to know the codebase well enough to be certain!

Setting state=closed when ssl is freed definitely feels like the way to go :-)
Thanks!!